dsa-23au/java-dsa/NMac-app/src/BubbleSortCSV.java

```diff
@@ -0,0 +1,96 @@
+// The purpose of this program:
+// Use the Bubble Sort algorithm to sort the CSV file
+
+// Bubble sort is a simple comparison-based sorting algorithm.
+// Its name derives from the way smaller elements "bubble" to the top of the list.
+// The basic idea behind bubble sort is to repeatedly step through the list,
+// compare adjacent items, and swap them if they are in the wrong order.
+// This process is repeated until no more swaps are needed, which means the list is sorted.
+
+
+package bubbleSortCSV;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class BubbleSortCSV {
+
+    // Class to represent each driving record headers from the CSV file
+    public static class DrivingRecord {
+        String time;
+        String date;
+        String elapsed;
+        String orig;
+        String dest;
+        String detourEnroute;
+        String driver;
+        String distance;
+
+        // Overriding toString to print each of the driving record columns
+        @Override
+        public String toString() {
+            return time + ", " + date + ", " + elapsed + ", " + orig + ", " + dest + ", " + detourEnroute + ", " + driver + ", " + distance;
+        }
+    }
+
+
+    // Main method of the program
+    public static void main(String[] args) {
+        // Read CSV file and get a list of driving records
+        List<DrivingRecord> records = readCSV("Time_Driving_Spreadsheet_test2.csv");
+
+        // Sort the records based on time
+        bubbleSort(records);
+
+        // Print sorted records
+        System.out.println("Sorted Records by Time:");
```

Comment:
This looks like a good start. The problem I see with sorting by `time` alone is that you will likely have more than one departure time on different days